.github/PULL_REQUEST_TEMPLATE/pull_request_template.md

```diff
@@ -30,3 +30,63 @@ Describe the grammar changes or new functions.
 
 Why problems is this JEP addressing?
 Has alternatives been considered?
+
+## Implementation Survey
+
+*Omit this section if not relevant*
+
+Provide a minimal demonstration of how the proposed functionality would be implemented for each programming language.

```

Comment:
Do you mean implemented as a library author ? Or used from a consuming programming language ?

I’m strongly in favor of providing a first implementation that supports syntax changes in a JEP. 
That’s [what I did](https://github.com/jdevillard/JmesPath.Net/pull/46) when I proposed the [JEP-14 Compact syntax for multi-select-hash](https://github.com/jmespath-community/jmespath.jep/pull/4).

However, I’m in no position to provide a, say, Lua or Ruby implementation.

Or are you suggesting we would need to gather community contributions from sufficient people knowledgeable enough in those programming languages before a JEP could be deemed sufficiently universal ?

Should this be relevant to syntax changes only, and (usually) not for function proposals ?

Reply:
As a library author. I think it needs to be **minimally** demonstrated that the proposed functionality can be implemented in each targeted language. Requiring a example in all languages ensures that research has been done into each language before the PR is accepted.

If someone isn't able to produce an example in select languages then it would be important to get input from someone who can.

I mostly intended this for function proposals. For example if we add a regex() function to the spec, what libraries are going to be added as dependencies to pull in regex support. If these are terribly burdensome dependencies for the jmespath library of that language to require, then we might want to reconsider adding it to the spec.